File: exercises/D309/solution.c
#include <stdio.h>
#include <string.h>
char s[101], t[101];
int main() {
    scanf("%s %s", s, t);
    int len1 = strlen(s);
    int len2 = strlen(t);
    int i;
    for (i = 0; i < len1; i++) {
        if (s[i] >= 'a') {
            s[i] -= 32;
        }
    }
    for (i = 0; i < len2; i++) {
        if (t[i] >= 'a') {
            t[i] -= 32;
        }
    }
    int a = strcmp(s, t);
    if (a < 0) {
        printf("Smaller\n");
    } else if (a == 0) {
        printf("Equal\n");
    } else {
        printf("Greater\n");
    }
    return 0;
}

File: exercises/D309/solution.pas
var
    s, t: string;
    len1, len2, i, a: longint;
begin
    readln(s);
    readln(t);
    len1 := length(s);
    len2 := length(t);
    for i := 1 to len1 do
        if (s[i] >= 'a') then
            s[i] := chr(ord(s[i]) - 32);
    for i := 1 to len2 do
        if (t[i] >= 'a') then
            t[i] := chr(ord(t[i]) - 32);
    if (s < t) then
        writeln('Smaller')
    else if (s = t) then
        writeln('Equal')
    else
        writeln('Greater')
end.
